lm_eval/tasks/ja/__init__.py

```diff
@@ -0,0 +1,34 @@
+import re
+
+
+class MecabTokenizer:
+    def __init__(self) -> None:
+        from fugashi import Tagger
+        self.tagger = Tagger('-Owakati')
+
+    def normalize_answer(self, text):
+        """Lower text and remove punctuation, articles and extra whitespace."""
+        import emoji
+        import neologdn
+
+        def white_space_fix(text):
+            return " ".join(text.split())
+        
+        def remove_emoji(text):
+            text = "".join(["" if emoji.is_emoji(c) else c for c in text])
+            emoji_pattern = re.compile(
+                "["
+                u"\U0001F600-\U0001F64F"  # emoticons
+                u"\U0001F300-\U0001F5FF"  # symbols & pictographs
+                u"\U0001F680-\U0001F6FF"  # transport & map symbols
+                u"\U0001F1E0-\U0001F1FF"  # flags (iOS)
+                u"\U00002702-\U000027B0"
+                "]+",
+                flags=re.UNICODE,
+            )
+            return emoji_pattern.sub(r"", text)
+        
+        return white_space_fix((neologdn.normalize(remove_emoji(text))))
```

Comment:
```suggestion
        text = remove_emoji(text)
        # see neologdn docs for details, but handles things like full/half width variation
        text = neologdn.normalize(text)
        text = white_space_fix(text)
        return text
```

Unpacked this a bit so it's easier to read. Also people may be unfamiliar with neologdn so it's helpful to mention why it's here.